pkg/nar/header.go

```diff
@@ -92,6 +92,8 @@ func (fi headerFileInfo) Mode() fs.FileMode {
 		mode |= (syscall.S_IXUSR | syscall.S_IXGRP | syscall.S_IXOTH)
 	case TypeSymlink:
 		mode = fs.ModePerm | fs.ModeSymlink
+	case TypeUnknown:
```

Comment:
Add a comment why this is impossible to construct through NAR reader, only by manually populating structs.

Also, please add a check for it to the `Header.Validate()` function, so it's possible to check without panic'ing.